### PR TITLE
refactor(VTextField): convert to ts

### PR DESCRIPTION
**Pull request by johnleider**
## Description
Converted VTextField to typescript. Updated maskable to extend Vue so it is a valid ts mixin.

## Motivation and Context
Convert framework to TS

## How Has This Been Tested?
jest

## Markup:
n/a

<details>

```vue
// Paste your FULL Playground.vue here
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VTextField/VTextField.ts [hunk 1]
```diff
@@ -204,8 +221,8 @@ export default VInput.extend({
       this.onFocus()
     },
     /** @public */
-    blur () {
-      this.$refs.input ? this.$refs.input.blur() : this.onBlur()
+    blur (e: Event) {
```
Comment (sh7dm) [inline packages/vuetify/src/components/VTextField/VTextField.ts]: BlurEvent?

Comment (sh7dm) [reply packages/vuetify/src/components/VTextField/VTextField.ts]: It's called `FocusEvent`

File: packages/vuetify/src/components/VTextField/VTextField.ts [hunk 2]
```diff
@@ -364,13 +381,13 @@ export default VInput.extend({
         this.suffix ? this.genAffix('suffix') : null
       ])
     },
-    genAffix (type) {
+    genAffix (type: string) {
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VTextField/VTextField.ts]: `type: 'append' | ’prepend'` (or whatever value it takes)

File: packages/vuetify/src/components/VTextField/VTextField.ts (hunk 1 above)
Comment (jacekkarczmarczyk) [reply packages/vuetify/src/components/VTextField/VTextField.ts]: Similar for onFocus, onMouseUp etc

**Pull request merged**